src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -27,6 +28,9 @@ import {
 })
 export class QuestionSubmissionFormComponent implements OnInit {
 
+  @ViewChild(ConstsumRecipientsQuestionConstraintComponent)
+  constSumConstraint: ConstsumRecipientsQuestionConstraintComponent | undefined;
```

Comment:
The task might be slightly difficult than you think, the current solution works but not the best I think. Firstly

- Violation: Event should be used for parent-child communication
- Violation: Not follow OCP principle. Image if we want to add validation and disable the button for other question types?

Reply:
> The task might be slightly difficult than you think, the current solution works but not the best I think. Firstly
> 
> * Violation: Event should be used for parent-child communication
> * Violation: Not follow OCP principle. Image if we want to add validation and disable the button for other question types?

I wanted to use `Event` at first, but then the question became "how often should the child be emitting an update"? Since the constraint component is separate from the question input component, you can't trigger on change? Unless the flow goes: OnChange, response input (child) emit to question component (parent) -> question component (parent) emit to constraint component (child) -> constraint to question -> question (child) to submission form (parent)

The above flow felt more convoluted than just having an optional `ViewChild`. But I see now that my current implementation is not extendable to include validation for other questions. I will try and think about how I can use `Event` with a consideration for extensibility.